### PR TITLE
Documentation Website

### LINKED ISSUE #217: No documentation available for chartjs-plugin-annotation
Why can I not find any documentation  on this library?
Or where is it hidden?
Searches take me to chart,js, which has a link to this github, but still no documentation.
Are we supposed to guess at the syntax?
Are we supposed to guess at the possible parameters and their possible values?
No documentation IS an issue.

### LINKED ISSUE #294: Please add a React sample v1.0.0-beta (ChartJs v3)!!!
How do I make the following work with this plugin using the following???? Can some one get this to work and create a PR to the samples folder? You'd be helping out MANY developers!

    "chart.js": "3.0.0-beta.7",
    "chartjs-adapter-moment": "^0.1.2",
    "chartjs-plugin-annotation": "1.0.0-beta",

```import React, { useEffect, useRef, useCallback } from 'react'
import { Chart as Chartjs } from 'chart.js'

import type {
  ChartOptions,
  ChartData,
  ChartType,
  Plugin,
  UpdateMode,
  DefaultDataPoint,
} from 'chart.js'

import { noop } from '../../utils/noop'

export interface IChartProps {
  chartId: string
  data: ChartData
  options: ChartOptions
  type: ChartType
  updateMode?: UpdateMode
  plugins?: Plugin[]
}

export const Chart = ({ data, options, type, plugins, updateMode, chartId }: IChartProps) => {
  const chartInstance = useRef<Chartjs<ChartType, DefaultDataPoint<ChartType>, unknown>>({
    update: noop,
    destroy: noop,
  } as Chartjs)

  useEffect(() => {
    chartInstance.current.options = options
    chartInstance.current.data = data

    chartInstance.current.update(updateMode)
  }, [data, options])

  const nodeRef = useCallback<(instance: HTMLCanvasElement | null) => void>((node) => {
    chartInstance.current.destroy()

    if (node) {
      chartInstance.current = new Chartjs(node, {
        type,
        data,
        options,
        plugins,
      })
    }
  }, [])

  return <canvas ref={nodeRef} id={chartId} />
}

Chart.register = Chartjs.register || noop
````

### LINKED ISSUE #336: Update the examples on the README
Hi!
The documentation is outdated, leading users of this lib into error. One example of this is the documentation regarding the text color. Following [this](https://github.com/chartjs/chartjs-plugin-annotation/blob/master/README.md#line-annotations) example on README, the color of the label doesn't change, but the example given on this [issue](https://github.com/chartjs/chartjs-plugin-annotation/issues/168#issuecomment-770325995) actually works. The `fontColor` property doesn't even appear on the README, and only on that issue.

I believe that the docs are outdated, and could be updated.

### LINKED ISSUE #343: Host documentation & samples
The samples and documentation should be hosted somewhere. We should be able to do this using GitHub pages similar to how https://github.com/chartjs/chartjs-chart-smith/ works where the docs end up at https://www.chartjs.org/chartjs-chart-smith/

### PR DESCRIPTION
**Pull request by etimberg**
Adds a doc site based on docusaurus. My thinking is that we deploy to the `gh-pages` branch and it will appear as https://www.chartjs.org/chartjs-plugin-annotation. This would be similar to the [Smith Chart](https://www.chartjs.org/chartjs-chart-smith/index) docs.

## To Do

- [x] Line annotation docs
- [x] Interaction docs
- [ ] Decide what to do with the samples

annotation docs

Resolves #343 
Resolves #336 
Resolves #294 
Resolves #217

### CODE REVIEW

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+jobs:
+  checks:
+    if: github.event_name != 'push'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+          run: |
```
Comment (kurkle) [inline .github/workflows/documentation.yml:18]: This might be incorrectly nested under setup-node

File: .github/workflows/documentation.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            npm run build
+  gh-release:
+    if: github.event_name != 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+      - name: Add key to allow access to repository
```
Comment (kurkle) [inline .github/workflows/documentation.yml:30]: Is this requirement somehow related to how `npx docusaurus deploy` works?

File: docs/docusaurus.config.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  title: 'chartjs-plugin-annotation',
+  tagline: 'Annotation plugin for Chart.js',
+  url: 'https://chartjs.org',
+  baseUrl: '/chartjs-plugin-annotation/',
+  favicon: 'img/favicon.ico',
+  organizationName: 'chartjs', // Usually your GitHub org/user name.
+  projectName: 'chartjs-plugin-annotation', // Usually your repo name.
+  plugins: [],
+  scripts: [
+    'https://cdn.jsdelivr.net/npm/chart.js@3.0.0-beta.13/dist/chart.js',
```
Comment (kurkle) [inline docs/docusaurus.config.js:12]: could use @next for the versions

File: docs/docusaurus.config.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
+// VERSION replaced by deploy script
```
Comment (kurkle) [inline docs/docusaurus.config.js:1]: this should be removed

File: docs/docusaurus.config.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+          title: 'Developers',
+          items: [
+            {
+              label: 'GitHub',
+              href: 'https://github.com/chartjs/chartjs-plugin-annotation',
+            },
+          ],
+        },
+      ],
+      copyright: `Copyright © ${new Date().getFullYear()} Chart.js contributors.`,
```
Comment (kurkle) [inline docs/docusaurus.config.js:37]: chartjs-plugin-annotation contributors?

**Review by kurkle — COMMENTED** [review]
Couple of comments. Great work!

**Comment by stockiNail** [comment]
> 
> 
> Couple of comments. Great work!

Really great! I think that was needed (in my opinion in all projects) because only the README is not enough when teh configuration and features are growing!

Looking forward to help if needed (as soon as I can.. :( ).

**Comment by stockiNail** [comment]
@etimberg @kurkle as promised in another PR/issue I had started creating a README with the tables of options (style CHART.JS) (see my branch https://github.com/pepstock-org/chartjs-plugin-annotation/tree/readme#global-options) but it was (and is) not enough.

**Comment by etimberg** [comment]
That's really helpful @stockiNail! I'll cross reference your tables against the docs as I write them. I think I'll be able to get the remaining doc pages done tonight.

File: docs/docusaurus.config.js (hunk 5 above)
Comment (LeeLenaleee) [inline docs/docusaurus.config.js:37]: Shouldn't the copyright year be the same as the one in the license instead of the current year?

Comment (kurkle) [reply docs/docusaurus.config.js]: I think both should be 2016-2021 currently

**Review by kurkle — APPROVED** [review]

File: docs/docusaurus.config.js (hunk 5 above)
Comment (etimberg) [reply docs/docusaurus.config.js]: I can update both

**Review by kurkle — APPROVED** [review]
